### DISCUSSION TITLE
Impossible to update Eclipse m2e due to problem of dependency (need javaSE 17; installed: 18)

### DISCUSSION THREAD

**Discussion by gprat31** [eclipse-m2e/m2e-core#1415] [Q&A]
When trying to update Eclipse m2e, I have the following problem:

```
Cannot complete the install because some dependencies are not satisfiable
  Software being installed: a.jre.javase 18.0.0
  Software currently installed: Eclipse IDE for Java Developers 4.27.0.20230309-1200 (epp.package.java 4.27.0.20230309-1200)

Cannot satisfy dependency:
    From: Eclipse IDE for Java Developers 4.27.0.20230309-1200 (epp.package.java 4.27.0.20230309-1200)
    To: osgi.ee; JavaSE [17.0.0,17.0.0]
```

My configuration is Linux (Ubuntu) and openjdk version "1.8.0_362" 

Thanks

**Comment by mickaelistria** [comment]
Can you please provide detailed steps to reproduce, starting from a fresh installation?

**Comment by gprat31** [comment]
- **_After my first tries, I remove  m2e in "Project nature"._**


![maven0](https://github.com/eclipse-m2e/m2e-core/assets/47821218/bf74b6cd-2b7d-4f5c-be6f-c555c2a05605)


- **_Then I uninstalled m2e and  I open MarketPlace to install it from sratch_**


![maven1](https://github.com/eclipse-m2e/m2e-core/assets/47821218/aa064c29-2b7f-4533-9b17-bf48350175c8)


- **_Then I let the selected items and confirm the installation_** 


![maven2](https://github.com/eclipse-m2e/m2e-core/assets/47821218/f8187078-a5c4-402c-ba3f-5760e330fd48)


- **_A popup tells me:_**


![maven3](https://github.com/eclipse-m2e/m2e-core/assets/47821218/b41a20ed-4668-4356-bc17-ebbc07741d58)


- **_As I need absolutely maven for my project, I proceed with the install ... And the error is:_**


![maven4](https://github.com/eclipse-m2e/m2e-core/assets/47821218/f94aef64-eb56-4f73-8977-285ba49387cd)


- **_I tried again by deselecting the non required item, but same result ..._**

**Comment by gprat31** [comment]
- **_If I open a closed maven project, as I uninstalled m2e, the following popup opens and the solution leeds to the marketplace shown above_**

![maven6](https://github.com/eclipse-m2e/m2e-core/assets/47821218/c79211f1-3b6b-43b0-a992-e205ef91fd16)

**Reply by mickaelistria** [reply]
The "Software being installed: a.jre.javase 18.0.0" seems rather strange because nothing requires `a.jre.javase` 18 from Marketplace.
Can you please try the alternative installation processes, without Marketplace client, that are described in https://github.com/eclipse-m2e/m2e-core#-installation ?

**Comment by gprat31** [comment]
- **_I tried the alternative installation. I selected all the items_** :

![Screenshot_2023-06-09_09-45-08](https://github.com/eclipse-m2e/m2e-core/assets/47821218/0fe4620a-85ee-4df4-b712-4fef62d342e9)

- **_some m2e part seems not compatible but the main (?) part are OK_** : 


![Screenshot_2023-06-09_09-46-17](https://github.com/eclipse-m2e/m2e-core/assets/47821218/eee33165-b3d5-448d-98cb-fdc6d1561861)


- **_the install details are_** : 

![Screenshot_2023-06-09_09-46-39](https://github.com/eclipse-m2e/m2e-core/assets/47821218/a37c390f-2c8d-4731-87f1-ebcfb5c3bcc5)

- **_After the installation, I checked what is installed_**:


![installed](https://github.com/eclipse-m2e/m2e-core/assets/47821218/508fc0f7-2657-499c-b650-9f9a1626b710)


- **_then I open a closed maven project (I ignored the "Marketplace solution" in order to use the m2e installed) but in the configure menu: maven not present ..._**


![configure](https://github.com/eclipse-m2e/m2e-core/assets/47821218/3aa2d806-6092-43e1-9cec-1c1fd620d7c0)


- **_when opening the properties of this project, in the nature part_**:

![missing](https://github.com/eclipse-m2e/m2e-core/assets/47821218/f014cabc-62c8-4d23-9406-f7f1b726c533)


- **_for the record, after these steps,  I checked the Marketplace solution (the version proposed of m2e is 2.3.0):_** 


![Screenshot_2023-06-09_10-09-53](https://github.com/eclipse-m2e/m2e-core/assets/47821218/b2cd2f32-630a-455e-9a43-05e5cc41634a)

**Reply by mickaelistria** [reply]
Can you please show what the installation wizard tells when you select "Show original error and build my own solution" ?

**Comment by gprat31** [comment]
When trying to install only: M2E Complete Development kit and M2E Integration fo Eclipse (I didn't select M2E PDE Integration/ POM editor / SLF4J) :

```
Cannot complete the install because of a conflicting dependency.
  Software being installed: M2E - Complete Development Kit 2.3.0.20230523-2106 (org.eclipse.m2e.sdk.feature.feature.group 2.3.0.20230523-2106)
  Software currently installed: M2E - POM Editor using LemMinX language server (includes Incubating components) 2.0.3.20230202-1631 (org.eclipse.m2e.lemminx.feature.feature.group 2.0.3.20230202-1631)
  Only one of the following can be installed at once: 
    M2E - POM Editor using LemMinX language server (includes Incubating components) 2.0.4.20230520-0934 (org.eclipse.m2e.lemminx.feature.feature.jar 2.0.4.20230520-0934)
    M2E - POM Editor using LemMinX language server (includes Incubating components) 2.0.3.20230202-1631 (org.eclipse.m2e.lemminx.feature.feature.jar 2.0.3.20230202-1631)
  Cannot satisfy dependency:
    From: M2E - POM Editor using LemMinX language server (includes Incubating components) 2.0.3.20230202-1631 (org.eclipse.m2e.lemminx.feature.feature.group 2.0.3.20230202-1631)
    To: org.eclipse.equinox.p2.iu; org.eclipse.m2e.lemminx.feature.feature.jar [2.0.3.20230202-1631,2.0.3.20230202-1631]
  Cannot satisfy dependency:
    From: M2E - POM Editor using LemMinX language server (includes Incubating components) 2.0.4.20230520-0934 (org.eclipse.m2e.lemminx.feature.feature.group 2.0.4.20230520-0934)
    To: org.eclipse.equinox.p2.iu; org.eclipse.m2e.lemminx.feature.feature.jar [2.0.4.20230520-0934,2.0.4.20230520-0934]
  Cannot satisfy dependency:
    From: M2E - Complete Development Kit 2.3.0.20230523-2106 (org.eclipse.m2e.sdk.feature.feature.group 2.3.0.20230523-2106)
    To: org.eclipse.equinox.p2.iu; org.eclipse.m2e.lemminx.feature.feature.group [2.0.4.20230520-0934,2.0.4.20230520-0934]
```

**Comment by gprat31** [comment]
When selecting all the 5 M2E items :

```
Cannot complete the install because of a conflicting dependency.
  Software being installed: M2E - POM Editor using LemMinX language server (includes Incubating components) 2.0.4.20230520-0934 (org.eclipse.m2e.lemminx.feature.feature.group 2.0.4.20230520-0934)
  Software currently installed: Wild Web Developer XML tools 1.0.2.202301201156 (org.eclipse.wildwebdeveloper.xml.feature.feature.group 1.0.2.202301201156)
  Only one of the following can be installed at once: 
    XML support using Language Server 1.1.0.202304242122 (org.eclipse.wildwebdeveloper.xml 1.1.0.202304242122)
    XML support using Language Server 1.0.2.202301201156 (org.eclipse.wildwebdeveloper.xml 1.0.2.202301201156)
  Cannot satisfy dependency:
    From: M2E Maven POM File Editor using Wild Web Developer, LemMinX and Maven LS extension (requires Incubating components) 2.0.3.20230520-0934 (org.eclipse.m2e.editor.lemminx 2.0.3.20230520-0934)
    To: osgi.bundle; org.eclipse.wildwebdeveloper.xml [1.1.0,1.2.0)
  Cannot satisfy dependency:
    From: M2E - POM Editor using LemMinX language server (includes Incubating components) 2.0.4.20230520-0934 (org.eclipse.m2e.lemminx.feature.feature.group 2.0.4.20230520-0934)
    To: org.eclipse.equinox.p2.iu; org.eclipse.m2e.editor.lemminx [2.0.3.20230520-0934,2.0.3.20230520-0934]
  Cannot satisfy dependency:
    From: Wild Web Developer XML tools 1.0.2.202301201156 (org.eclipse.wildwebdeveloper.xml.feature.feature.group 1.0.2.202301201156)
    To: org.eclipse.equinox.p2.iu; org.eclipse.wildwebdeveloper.xml [1.0.2.202301201156,1.0.2.202301201156]
```

**Reply by mickaelistria** [reply]
Great, thanks. So please make sure you're also trying to upgrade Wild Web Developer from https://download.eclipse.org/wildwebdeveloper/releases/latest either in a 1st step or at the same time.

**Comment by gprat31** [comment]
I installed at first step Wild Web Developer from the above URL, and everything runs well including the update of m2E.
After Eclipse restart, but still the closed project once re-open is not recognized as maven. Even in the nature of the project, even when trying to add a new nature : maven does not exists ...

![nature](https://github.com/eclipse-m2e/m2e-core/assets/47821218/1206a1d4-fb9c-4e8f-87d6-f04d6a1a6fc2)


So I uninstalled Wild Web Developer and then also uninstall m2e (all the items named m2e).
I installed again m2e manually as you explained ... the install went well. But I noticed that now the version is 2.3 ...
But still the project once closed and re-opened is not recognised as maven (same tests as above) :-(

**Comment by laeubi** [comment]
Maybe just wait a few days until the new 2023-06 is promoted and try with a fresh install e.g. Eclipse for RCP/RAP developers?

**Comment by gprat31** [comment]
I will wait so hopping it'll solve the problem !

**Comment by gprat31** [comment]
Problem solved:
I installed from scratch eclipse 2023-03 with the installer (https://www.eclipse.org/downloads/)
and in the installer, chose the JRE 18.0.2.1 
Then I opened Eclipse : my projects were recognized as maven projects !!!
